jax/api.py

```diff
@@ -1967,6 +1967,61 @@ def device_put(x, device: Optional[xc.Device] = None):
   return tree_map(lambda y: xla.device_put_p.bind(y, device=device), x)
 
 
+def device_put_sharded(x: Sequence[Any], devices: Sequence[xc.Device]) -> pxla.ShardedDeviceArray:
+  """Shards the input to specified devices, returning ShardedDeviceArrays.
```

Comment:
```suggestion
  """Transfers pre-sharded input to the specified devices, returning ShardedDeviceArrays.
```

This comment threw me off initially. It's not actually sharding the input, it's assembling ShardedDeviceArrays from already-sharded data.

I would also consider renaming the function, maybe `multi_device_put`? I'm ok with the current name, but it still makes me think it's gonna do the sharding itself (that might be my prior bias though).